### PR TITLE
MailBox Page is fully functional (except read receipts)

### PR DESCRIPTION
**Pull request by mrkarthik3**
1. Fixed bugs & made improvements in sub-components needed in MailBox.
2. Added Route for MailBox on Left SideBar.
3. MailBox page is created.

image

When interest is sent... (Vishal is interest sender, Shivani is interest receiver)
In Sender's mailbox... it shows like below...
image

When interest is received, Receiver's mailbox will show that interest like below
with options to accept/reject that interest.
image

When Receiver accepts interest, Sender & Receiver both... can see "View & Send Messages" Button and can freely communicate.
When interest is accepted, it shows "View & Send Messages" Button
image
image

When interest is rejected, it shows different messages based on who sent the interest...

Interest sender will get this message & option to delete the interest.
image

Interest receiver... who rejected the interest will see the following along with option to delete the interest.
image

Old messages can be read via modal 
New messages are sent via the same modal.
image

As usual, there are notifications.... like this (wherever relevant).
image

Confirmation after sending message.
image

### CODE REVIEW

File: apps/pm-user/src/components/molecules/interestBox/InterestBox.js [hunk 1]
```diff
@@ -179,34 +177,34 @@ const InterestBox = ({
...
-  isRejected: false,
-  conversations: [],
-};
+// InterestBox.propTypes = {
+//   idOfLoggedInUser: PropTypes.string,
+//   interestSenderName: PropTypes.string,
+//   interestSenderId: PropTypes.string,
+//   interestSenderAge: PropTypes.number,
+//   interestSenderImage: PropTypes.string,
+//   interestReceiverId: PropTypes.string,
```
Comment (developervinit) [inline apps/pm-user/src/components/molecules/interestBox/InterestBox.js:186]: alphabetical order missing here

File: apps/pm-user/src/pages/mailBox/MailBox.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      { messageSenderId: 'shivani', messageReceiverId: 'vishal', message: 'Hello, Vishal.' },
+      { messageSenderId: 'vishal', messageReceiverId: 'shivani', message: 'Dummy message from Vishal.' },
+      { messageSenderId: 'vishal', messageReceiverId: 'shivani', message: 'Dummy message from Vishal.' },
+      { messageSenderId: 'shivani', messageReceiverId: 'vishal', message: 'Dummy message from Shivani.' },
+      { messageSenderId: 'shivani', messageReceiverId: 'vishal', message: 'Dummy message from Shivani.' },
+    ],
+  },
+];
+
+const MailBox = ({ interestsSentAndReceived, idOfLoggedInUser }) => {
```
Comment (developervinit) [inline apps/pm-user/src/pages/mailBox/MailBox.js:29]: props should be in alphabetical oreder

File: apps/pm-user/src/pages/mailBox/MailBox.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      </div>
+      <div className={styles.content}>
+        {renderMailBox}
+      </div>
+    </div>
+  );
+};
+
+MailBox.propTypes = {
+  interestsSentAndReceived: PropTypes.array,
```
Comment (developervinit) [inline apps/pm-user/src/pages/mailBox/MailBox.js:64]: propsType and default-porps should be in alphabetical order

Comment (mrkarthik3) [reply apps/pm-user/src/pages/mailBox/MailBox.js]: Ok. Corrected them

File: apps/pm-user/src/pages/mailBox/MailBox.js (hunk 2 above)
Comment (mrkarthik3) [reply apps/pm-user/src/pages/mailBox/MailBox.js]: Ok. Corrected them.

File: apps/pm-user/src/components/molecules/interestBox/InterestBox.js (hunk 1 above)
Comment (mrkarthik3) [reply apps/pm-user/src/components/molecules/interestBox/InterestBox.js]: Ok. Corrected them.

**Pull request merged**